test/manipulation.test.js

```diff
@@ -875,6 +877,76 @@ describe('manipulation', function () {
         e.should.be.eql(new Error('Invalid date: X'));
       }
     });
+  });
+
+  describe('update/updateAll', function() {
+    var connectors = ['memory', 'transient'];
 
+    function getModel(connector) {
+      var ds = new DataSource(connector);
+      return ds.define('Model');
+    }
+
+    it('should implement the `updateAll` function', function() {
+      connectors.forEach(function(connector) {
+        var Model = getModel(connector);
+        Model.update.should.be.a.Function;
+        Model.updateAll.should.be.a.Function;
+      });
+    });
+
+    it('should not update the model when no match is found', function(done) {
+      // skip the transient connector
+      var index = connectors.indexOf('transient');
+      connectors.splice(index, 1);
+
+      async.each(connectors, function(connector, callback) {
```

Comment:
No no, you misunderstood me. Tests in this file don't deal with different connectors, they just provide a test using `global.getDataSource`. Each connector then includes this file in its test-suite and provides a correct implementation of `global.getDataSource` that returns a data-source backed by that connector.

So don't iterate over all connectors here, just run the test code.

For  the future reference, if we were note removing the loop, then it's better to move the loop *outside* the test and define one test per each item looped over, using a unique test name including the name of the item (connector). The current implementation gives useless assertion errors - it will tell you that `count is not exactly 0 but 1 instead`, but you have no idea which connector caused the failure.